src/packet/buffer_rx.rs

```diff
@@ -38,6 +43,14 @@ impl Depacketized {
             .expect("a depacketized to consist of at least one packet")
     }
 
+    pub fn first_sender_timings(&self) -> Option<(MediaTime, u32)> {
+        self.meta
+            .iter()
+            .min_by_key(|m| m.received)
+            .map(|m| m.last_sender_timing)
+            .expect("a depacketized to consist of at least one packet")
+    }
```

Comment:
Don't we just want the latest value? Why is this `first` (i.e. `min`)?

Reply:
it's picking the first last_sender_timing of the `first` packet that generated this sample.
There would be no difference if we picked the `last` (but its better than picking `any`).

In both cases it can be slightly wrong because it is possible that
- we receive the 1st packet of frame 3 and we enrich it with the last sender info
- we receive an updated sender info
- we receive the 1st packet of frame 2 and we enrich it with more recent sender info

=> In this case the sample-based api would flush out in-order samples with out-of-order sender infos

It's better to pick up the last value when MediaData is emitted or generated

Reply:
My thinking is like this, a frame is packetized into many packets that are typically all sent at the same time. They all have the same RTP timestamp. By picking the sender info based on the first packet that arrived we maximise the chance that we use values from an SR prior to the frame being sent. As Davide outlined it's still possible that this isn't the case, but it's minimised.

For exmaple:

```
    ┌──────┐
    │Frame │
    ├──────┘─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
       ┌──┐  ┌──┐ ┌──┐         ┌──┐      │
    │  │P2│  │P1│ │P3│         │P4│
       └──┘  └──┘ └──┘         └──┘      │
    └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
┌───┐           ┌───┐
│SR1│           │SR2│
└───┘           └───┘
```

Here we'd use the information from SR1 because that was the most recent SR when the first packet, P2, was received.